### ISSUE TITLE
How to query transactions inside a created_date range

### ISSUE THREAD

**Issue by victorjtfranco** [pagarme/pagarme-ruby#54]
**TL;DR:**
  - How do I query for transactions inside a `date_created` range ?
  - Is there an issue using `RestClient::ParamsArray` ?

I am trying to do the equivalent to
```shell
curl -X GET https://api.pagar.me/1/transactions \
-d 'api_key=ak_test_grXijQ4GicOa2BLGZrDRTR5qNQxJW0'
-d 'date_created=>=1483236000000'
-d 'date_created=<=1484689847590'
```
(code from https://docs.pagar.me/v2017-07-17/reference#retornando-transações)

using this gem. In order to filter in a range, I have found about `RestClient::ParamsArray`
> New in 2.0: The new RestClient::ParamsArray class allows callers to provide ordering even to structured parameters. This is useful for unusual cases where the server treats the order of parameters as significant or you want to pass a particular key multiple times.

But the gem seems to ignore params given on a `ParamsArray`:
```ruby
 PagarMe::Transaction.find_by(
  params: RestClient::ParamsArray.new([
  [:date_created,'>=1502401722'],
  [:date_created,'<=1502402530'],
  [:page,1],[:count, 9]]) 
).count
=> 10
# count is 10, I requested 9
```

Looking for the cause, this line caught my attention: https://github.com/pagarme/pagarme-ruby/blob/master/lib/pagarme/model.rb#L57

So, I think my `ParamsArray` is being ignored, and haven't found a better solution.

**Comment by leonardo-barbosa** [comment]
Hi @victorjtfranco , how you doing?

My name is Leonardo, I am part of the Support Team of Pagar.me.

I've been studying to recreate the query you tried with our ruby library, but couldn't.
Instead there's a alternative solution that you could do with Ruby `RestClient`, like the code below:

```ruby
response = RestClient.get 'https://api.pagar.me/1/transactions', {
params: RestClient::ParamsArray.new([
        [:api_key, 'YOUR_API_KEY',
        [:date_created,'>=1505444400000'],
        [:date_created,'<=1505876340000'],
        [:page,1],[:count, 10]
])}

response = JSON.parse(response)

response.each do |trx|
        puts trx
end
```
I hope this helps you, and in the meantime i'll be looking for a way to use our library to query params in a range.

Let me know if I can help you in anything else.

Best regards.

**Comment by murilohns** [comment]
hello @victorjtfranco, i'll close this issue, but if u need, feel free to reopen it :smile:
